server/src/GraphqlHTTPRequest.ts

```diff
@@ -1,10 +1,27 @@
 import { DocumentNode } from 'graphql';
 
 import { httpRequest } from './HTTPRequest';
+import { UPDATE_INVESTIGATION_METADATA } from './DBService/InvestigationMetadata';
 
 export const graphqlURL = '/coronai/graphql';
 export const baseUrl = 'http://localhost:';
 
-export const graphqlRequest = (query: DocumentNode, variables?: any) => (
+export const graphqlRequest = (query: DocumentNode, requestHeaders: any, variables?: any) => (
     httpRequest(baseUrl+process.env.PORT+graphqlURL, 'POST', {query: query.loc?.source.body, variables: { ...variables}})
+    .then((result) => {
+        if (requestHeaders) {
+            //@ts-ignore
+            if (query.definitions[0].operation === 'mutation') {
+                httpRequest(baseUrl+process.env.PORT+graphqlURL, 'POST', {
+                    query: (UPDATE_INVESTIGATION_METADATA as DocumentNode).loc?.source.body,
+                    variables: {
+                        epidemiologyNumber: +requestHeaders.epidemiologynumber,
+                        lastUpdateTime: new Date(),
+                        lastUpdator: requestHeaders.username
+                    }
+                });
```

Comment:
pass to func for readability and testability

Reply:
also username would be present in the request.locals in the future, and not in the headers

Reply:
you could consider using a update_investigation_metadata express-middleware to handle this issue